templates/app-ts/@types/index.d.ts

```diff
@@ -0,0 +1,12 @@
+import * as http from 'http'
+import fastify from 'fastify'
+
+declare module 'fastify' {
+  export interface FastifyInstance<
+  HttpServer = http.Server,
+  HttpRequest = http.IncomingMessage,
+  HttpResponse = http.ServerResponse
+> {
+    someSupport(): string;
+  }
+}
```

Comment:
What is this file for?

Reply:
Is the extension of the FastifyInstance in order to add the custom plugin to it.
Without it TS will throw an error because the type FastifyInstance has no property called someSupport.

Reply:
Is this customary to put it in a @types folder? can it leave in the same file of the plugin?

Reply:
It is by preference. My preference is that types are inlined as thats the most definite way for them to be included in a TS project. This method is valid, but requires users to configure their project to look for them in this directory.